### DISCUSSION TITLE
Render jinja template in a standalone class

### DISCUSSION THREAD

**Discussion by priyankagovindraj** [apache/airflow#43805] [Q&A]
Hi,

I have a use case:
ex_task = DummyOperator(
task_id = task_id,
dag=dag,
checks=
SomeCheckClass(
table=SqlTable("templated_table", "another_templated_field", "just_a_string",...))
]
)

SqlTable is a standalone class which has the templated fields. I have tried to get these fields rendered in the upstream base classes, however I have not been able to get it resolved.
Any ideas on how I can get this done?

**Comment by andresitorusz** [comment]
Could you provide the SqlTable code snippet? Focus on how you handle those jinja templates.

**Reply by priyankagovindraj** [reply]
`
class SqlTable:
    name: Union[str, TableMacro]
    database: str
    data_source: DataSource
    partition_clause: Optional[str] = None
`
The name and partition_clause fields are templated

**Reply by priyankagovindraj** [reply]
I have also used template_fields in all the upstream classes.
